ddtrace/debugging/_function/discovery.py

```diff
@@ -155,7 +155,14 @@ def _collect_functions(module):
         for k, o in c:
             if PY2 and _isinstance(o, UnboundMethodType):
                 o = o.__func__
-            if _isinstance(o, (FunctionType, FunctionWrapper)) and abspath(o.__code__.co_filename) == path:
+
+            try:
+                code = o.__code__ if _isinstance(o, (FunctionType, FunctionWrapper)) else None
```

Comment:
maybe worth just doing `getattr` instead of the try except? Exception handling has a decent amount of overhead